soc/riscv/riscv-ite/it8xxx2/soc.c

```diff
@@ -191,8 +191,18 @@ void riscv_idle(enum chip_pll_mode mode, unsigned int key)
 	chip_pll_ctrl(mode);
 
 	do {
-		/* Wait for interrupt */
-		__asm__ volatile ("wfi");
+#ifdef CONFIG_I2C_IT8XXX2_CQ_MODE
+		/*
+		 * The EC processor(CPU) cannot be in the k_cpu_idle() during
+		 * the transactions with the CQ mode(DMA mode). Otherwise,
+		 * the EC processor would be clock gated.
+		 */
+		if (!i2c_enhance_cpu_idle_not_allowed())
```

Comment:
Seems better to put this at the top of the function and early-return if not allowed, because there's no reason to do the other work this function does if it's not going to `wfi`.

Reply:
This function is put here because it is most similar to the sequence of the original program. When the program goes to `wfi`(or is blocked by the function of cpu_idle_not_allowed()), it will jump out of the do-while loop if there is any interrupt.

Reply:
It still seems better to early-return, since `arch_interface.h` says `arch_cpu_idle` is permitted to return immediately if a low-power idle is not available and that will allow the system to do any other idle-level work instead of spinning here.

Reply:
Done.